### PR TITLE
Merge kotlinx-coroutines-core and kotlinx-coroutines-jdk8 modules

### LINKED ISSUE #3268: Merge kotlinx-coroutines-core and kotlinx-coroutines-jdk8
We're gradually moving away from 1.6 support (#1589), it seems to be the time to merge supplementary jdk8 artifact into the core.

To keep Android API at a reasonable level, jdk8 will be existing in a separate source-set that will depend on core's main, but not the vice versa. Additionally, we'll keep releasing empty `jdk8` artifact for compatibility reasons

### PR DESCRIPTION
**Pull request by qwwdfsad**
Fixes #3268

### CODE REVIEW

File: kotlinx-coroutines-core/build.gradle [hunk 1]
```diff
@@ -104,13 +128,36 @@ kotlin {
...
+    }
+
+    jvm {
+        def main = compilations.main
+        main.source(jvmCoreMain)
+        main.source(jdk8Main)
+
+        /* Create compilation for jvmCore to prove that jvmMain does not rely on jdk8 */
+        compilations.create('CoreMain') {
+            /* jvmCore is automatically matched as 'defaultSourceSet' for the compilation */
```
[the remark this answers is not in the document]
Comment (qwwdfsad) [reply kotlinx-coroutines-core/build.gradle:148]: Yes, commited manually

File: integration-testing/src/withGuavaTest/kotlin/Jdk8InCoreIntegration.kt [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+ */
+class Jdk8InCoreIntegration {
+
+    @Test
+    fun testFuture() = runBlocking<Unit> {
+        val future = future { yield(); 42 }
+        future.whenComplete { r, _ -> assertEquals(42, r)  }
+        assertEquals(42, future.await())
+    }
+}
```
[the remark this answers is not in the document]
Comment (qwwdfsad) [reply integration-testing/src/withGuavaTest/kotlin/Jdk8InCoreIntegration.kt:21]: Added a comment to Gradle configuration -- I'm just piggybacking on this source-set

File: kotlinx-coroutines-core/build.gradle [hunk 3]
```diff
@@ -104,13 +128,36 @@ kotlin {
...
+            tasks.getByName('check').dependsOn(compileKotlinTaskProvider)
+        }
+    }
+}
+
+kotlin {
     jvm {
```
[the remark this answers is not in the document]
Comment (dkhalanskyjb) [reply kotlinx-coroutines-core/build.gradle:155]: It's not clear in the GitHub interface now that the comment is about the lines
```
    }
}

kotlin {
    jvm {
```

[the remark this answers is not in the document]
Comment (qwwdfsad) [reply kotlinx-coroutines-core/build.gradle:155]: Probably not, merged these and checked that animal sniffer works (including the case where I remove `IgnoreJreRequirement`)

**Comment by dkhalanskyjb** [comment]
To spare you the wait: the files in `withGuavaTest` also need to be moved.

**Review by dkhalanskyjb — APPROVED** [review]

**Pull request merged**